common/components/org.wso2.carbon.data.agents.log4j/src/main/java/org/wso2/carbon/data/agents/log4j/appender/LogEventAppender.java

```diff
@@ -106,7 +109,13 @@ private String getStacktrace(Throwable e) {
     }
 
     public void close() {
-
+        try {
+            if(dataPublisher!=null){
+                dataPublisher.shutdown();
+            }
+        } catch (DataEndpointException e) {
+            log.error("Error in shutting down the data publisher " + e.getMessage(), e);
+        }
     }
 
```

Comment:
Hi Jasintha,

This close() method implement from the 'org.apache.log4j.Appender', hence we cant return Boolean or exception since it will effect to the method signature. So we are log the errors occurred while releasing the resources used by the 'LogEventAppender' class.

Thanks,
SajithD